src/TimeSpans.jl

```diff
@@ -242,12 +237,7 @@ julia> index_from_time(100, TimeSpan(Second(3), Second(6)))
 """
 function index_from_time(sample_rate, span)
     i = index_from_time(sample_rate, start(span))
-    j, is_rounded = index_and_is_rounded_from_time(sample_rate, stop(span))
-    # if `j` has been rounded down, then we are already excluding the right endpoint
-    # by means of that rounding. Hence, we don't need to decrement here.
-    if i != j && !is_rounded
-        j -= 1
-    end
+    j = index_from_time(sample_rate, stop(span) - Nanosecond(1))
```

Comment:
I *think* I am somehow doing something similar in AlignedSpans. There, instead of of `is_rounded=!isinteger(index)`, I am checking if the "error" is zero:

https://github.com/beacon-biosignals/AlignedSpans.jl/blob/21d2e331f0d861ef6bed35236b550fb688a23459/src/time_index_conversions.jl#L25 

https://github.com/beacon-biosignals/AlignedSpans.jl/blob/21d2e331f0d861ef6bed35236b550fb688a23459/src/interop.jl#L22-L25

If I make that change here, e.g. on `main` with
```
# Helper to get the index and whether or not it has been rounded
function index_and_is_rounded_from_time(sample_rate, sample_time::Period)
    time_in_nanoseconds = convert(Nanosecond, sample_time).value
    time_in_nanoseconds >= 0 || throw(ArgumentError("`sample_time` must be >= 0 nanoseconds"))
    time_in_seconds = time_in_nanoseconds / NS_IN_SEC
    index = time_in_seconds * sample_rate + 1
    return floor(Int, index), !iszero(time_from_index(sample_rate, index) - sample_time)
end
```
Then I get similar results to this PR. This seems more direct but I don't quite follow it.